api/v1/flow/flow.proto

```diff
@@ -12,8 +12,12 @@ option go_package = "github.com/cilium/cilium/api/v1/flow";
 
 message Flow {
     google.protobuf.Timestamp time = 1;
-    Verdict verdict = 2;
 
+    // identifier is a unique identifier for this flow. Currently an UUIDv4 but
+    // not guaranteed to stay that way.
+    string identifier = 34;
```

Comment:
Note to reviewers: I went with "identifier" because "id" is already overloaded (security identity, endpoint identity) and didn't want to commit to "uuid" since we might change the representation at some point.

Reply:
I am a bit worried about [Hyrum's Law](https://www.hyrumslaw.com/) here. If this field stays in long enough, it will be hard to change to a different identifier format.

Therefore I think it might be better to just name it `uuid` (which I think more clearly distinguishes it from id/identifier/flow-hash and thus is the better name) and assume/assert it's going to be a version of UUID. If we ever need to make more fundamental changes to the format, I expect we have to introduce a new identifier field anyways.

Reply:
Fair, I'll update the PR accordingly.

Reply:
@gandro done, please take another look :pray: